### PR TITLE
MGDAPI-4923 - generalise index building script

### PR DESCRIPTION
**Pull request by adam-cattermole**
# Issue link
[MGDAPI-4923](https://issues.redhat.com/browse/MGDAPI-4923)

# What
Generalise the index building script and allow it to work for all 3 products (3scale/rhsso/mcg)

# Verification steps
Ensure that you can successfully build index images for all three products. You can verify the bundle chain in the index images using the script from @MStokluska

```
get_bundle_chain(){
        INDEX_IMAGE=${1}
        DOCKER_IMAGE=$(docker run -d -p 50051:50051 $INDEX_IMAGE)
        grpcurl -plaintext localhost:50051 api.Registry/ListBundles | jq '{packageName: .packageName, name: .csvName, version: .version, bundlePath: .bundlePath, replaces: .replaces, skipRange: .skipRange, channelName: .channelName, dependencies: .dependencies}'
        docker kill $DOCKER_IMAGE
}
```

### CODE REVIEW

**Comment by adam-cattermole** [comment]
I am undecided whether this should be merged to the GCP branch or master - whilst MCG will be used for GCP, the changes here are unrelated to the code changes required to add MCG to RHOAM - open to reviewer thoughts.

**Comment by adam-cattermole** [comment]
This will require the bundle file for MCG to be added once the pipeline is merged and we have re-built the required bundles.

**Comment by valerymo** [comment]
## Important: need to be Merged together with https://gitlab.cee.redhat.com/integreatly-qe/ci-cd/-/merge_requests/797
Merge together with cicd MR required because of changes in RHSSO bundles and index creation.
## Validation 
Validation done for All products - 3scale, rhsso, mcg in pipelines:
- https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/job/Playground/job/valerym/job/build-product-index-4923-hhsso-1/  
- https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/job/Playground/job/valerym/job/build-product-index-4923-3scale-1/
- https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/job/Playground/job/valerym/job/build-mcg-bundle-4923-mcg-1/

## Lgtm 
Suggested for /lgtm

**Comment by adam-cattermole** [comment]
Added commit 8a9b2cd containing new mcg-bundles and rebased on master

File: Makefile [hunk 1]
```diff
@@ -318,11 +318,15 @@ create/olm/bundle:
 
 .PHONY: create/3scale/index
 create/3scale/index:
-	./scripts/create-3scale-index.sh
+	PRODUCT=3scale ./scripts/create-3scale-index.sh
```
Comment (valerymo) [inline Makefile]: Maybe we need more generic name of the script? like `create-index-image.sh` or something like this?
PRODUCT=3scale ./scripts/create-3scale-index.sh
PRODUCT=rhsso ./scripts/create-3scale-index.sh
PRODUCT=mcg ./scripts/create-3scale-index.sh

**Pull request merged**